File: frontend/src/services/ApplicationService.js
import api from "../api/api";

/* ---------- STUDENT ---------- */

export const applyForJob = (jobId) =>
    api.post(`/applications/apply/${jobId}`);

export const getMyApplications = () =>
    api.get("/applications/my");

export const getMyRecentApplications = () =>
    api.get("/applications/my/recent");

/* ---------- ADMIN ---------- */

export const getAllApplications = () =>
    api.get("/applications");

export const getRecentApplications = () =>
    api.get("/applications/recent");

export const getApplicationById = (id) =>
    api.get(`/applications/${id}`);

export const updateApplicationStatus = (id, status) =>
    api.put(`/applications/${id}/status?status=${status}`);

export const deleteApplication = (id) =>
    api.delete(`/applications/${id}`);

File: src/main/java/com/placement/portal/service/ApplicationService.java
package com.placement.portal.service;

import java.util.List;

import com.placement.portal.dto.ApplicationResponseDTO;

public interface ApplicationService {

    ApplicationResponseDTO applyJob(String studentEmail, Long jobId);

    List<ApplicationResponseDTO> getMyApplications(String studentEmail);

    List<ApplicationResponseDTO> getMyRecentApplications(String studentEmail);

    List<ApplicationResponseDTO> getAllApplications();

    ApplicationResponseDTO getApplicationById(Long applicationId);

    ApplicationResponseDTO updateApplicationStatus(
            Long applicationId,
            String status);

    void deleteApplication(Long applicationId);

    List<ApplicationResponseDTO> getRecentApplications();

}